redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -290,4 +292,20 @@ object Output {
       }
   }
 
+  case object KeyLongValueOutput extends Output[Map[String, Long]] {
+    protected def tryDecode(respValue: RespValue): Map[String, Long] = {
+      val mapOutput = KeyValueOutput.unsafeDecode(respValue)
```

Comment:
Good direction, but you cannot reuse `KeyValueOutput` here, because `KeyValueOutput` expects `RespValue.BulkString` for the value format, while your type should expect `RespValue.Integer`. `RespValue`s are protocol-specific values, so you cannot first parse something as `RespValue.BulkString` (in scala terms `String`), and then cast it to `RespValue.Integer`. You'll have to implement `tryDecode` directly (same logic as `KeyValueOutput`, but instead of `RespValue.BulkString`, for the second value, you should use `RespValue.Integer`).

Reply:
i see i will write it whole then